lambda/proxy.test.ts

```diff
@@ -124,7 +124,6 @@ describe('proxy', () => {
       ...baseEvent,
       headers: {
         ...baseEvent.headers,
-        'content-type': undefined,
```

Comment:
This key now _fails_ the test because our validation schema expects the value to be exactly `application/json` or `application/x-www-form-urlencoded'. If this header is undefined I think we would want the proxy to throw an error

Reply:
😨 I see why you did this now! I did not realize `content-type` was already in the `baseEvent` object

Reply:
And I now understand the `delete` thing you asked about. The problem with `delete` is that it would screw up other tests

Reply:
I just tested this in a console and I'm not sure it's fully effective:

```
const originalObject = {
  key1: 'value1',
  key2: 'value2',
};

const modifiedObject = {
  ...originalObject,
  key1: undefined,
};

console.log(modifiedObject);
```

```
{ key1: undefined, key2: 'value2' }
```

So it may be necessary to use some variation of `delete headers['content-type'];`

Reply:
I found a better way:

```
const { 'content-type', ...headersWithoutContentType } = baseEvent.headers;

const event: APIGatewayProxyWithLambdaAuthorizerEvent<any> = {
  ...baseEvent,
  headers: {
    ...headersWithoutContentType,
    'Content-Type': 'application/json'
  },
  body: JSON.stringify(VALID_PUSH_PAYLOAD),
  pathParameters: {
    endpointId
  }
};
```

Reply:
The way you wrote the test is really the best way. The tests passed in my first attempt because I made the test wrong and implemented wrong. Now I implemented correctly and the test (with no change) passes!

Reply:
I did a little more testing and I think the way you've rewritten the headersSchema function does require changes to the tests to get a truly valid result.

Right now the only reason `Content-Type` `applicaton/json` value is asserting itself is because it is accessed by the headersSchema method second, and thus its value overwrites the `undefined` value for the `content-type` key. I do not know if that order can always be relied upon.

Previously the way I was setting `content-type` to `undefined` works reliably because of this line:

```
const contentType = headersResult['content-type'] || headersResult['Content-Type'];
```

The `content-type` will resolve to undefined, so `Content-Type` is used instead, and this is a reliable comparison that won't change based on order the key/value pair is processed.

Reply:
So to confirm I have the correct understanding, I ran just that one `Content-Type` test and logged the value of `mapKeys(obj, (_, key) => key.toLowerCase())` inside the [preprocess](https://github.com/colinhacks/zod#preprocess) method, and got
```ts
{
  accept: '*/*',
  'content-type': 'application/json',
  host: 'some-api.us-west-2.amazonaws.com'
}
```
So it is naturally sanitizing the object of any keys with values of `undefined`. When zod performs the actual validation using 
```ts
z.object({
  'content-type': z.nativeEnum(CONTENT_TYPES)
})
```
it will reject `'content-type': undefined`, so if the test passes it means that it's not seeing that. With your change from #134, it may have allowed that since the schema was `.optional()`

In any case, I updated the test to be more clear so there's no ambiguity whatsoever, and so it mimics a real headers object that the proxy would actually receive.